### ISSUE TITLE
`npm run prune-storage` doesn't respect custom environment variables

### ISSUE THREAD

**Issue by decentral1se** [Chocobozzz/PeerTube#4144]
**Describe the current behavior**

I am running `npm run prune-storage` within the context of docker where the postgres database is in another container at the hostname `db`. When I run `npm run prune-storage` I see the following logs. The script is trying to connect to my localhost but that is not the right IP.

```
  original: Error: connect ECONNREFUSED 127.0.0.1:5432
      at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1144:16) {
    errno: 'ECONNREFUSED',
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',
    port: 5432
  }
}
```

Here is my env:

```
root@f9c71fb6bc33:/app# env | grep -i db
PEERTUBE_DB_HOSTNAME=db
PEERTUBE_DB_PASSWORD=...
PEERTUBE_DB_USERNAME=peertube
```

**Steps to reproduce:**

1. Deploy peertube using docker (e.g. https://git.autonomic.zone/coop-cloud/peertube)
2. Run `npm run prune-storage`

**Describe the expected behavior**

Pruning succeeds.

**Additional information**

* PeerTube instance:
  * URL: https://tv.lumbung.space
  * version: `chocobozzz/peertube:v3.1.0-buster`
  * NodeJS version: `v12.21.0`
  * Ffmpeg version: `4.1.6-1~deb10u1`

**Comment by Chocobozzz** [comment]
Hello,

Does `docker-compose exec -u peertube peertube npm run prune-storage` work?

**Comment by decentral1se** [comment]
Hey, no, that command doesn't work. It fails with the error above. Please note I am using a docker swarm deployment, so I run commands directly in the container. But that uses the peertube user so I think the effect is the same. My configuration is here https://git.autonomic.zone/coop-cloud/peertube/src/branch/main/compose.yml

**Comment by decentral1se** [comment]
I gues I am wondering how does the `prune-storage.js` script read database connection details? Not from the environment or?

**Comment by decentral1se** [comment]
Aha I believe it has to do with https://github.com/Chocobozzz/PeerTube/blob/develop/support/docker/production/config/custom-environment-variables.yaml. So, I am using the environment to configure my iinstance: https://git.autonomic.zone/coop-cloud/peertube/src/commit/f9b826d2f8e308298472bf39dc1c371482be7095/compose.yml#L38-L49. Therefore, there is no `production.yml` to read when I specify `export NODE_ENV=production` in the `/app/config/...` directory. Can the scripts also read from the custom environment variables?

**Comment by Chocobozzz** [comment]
> Can the scripts also read from the custom environment variables?

Yes, you should be able to use https://github.com/Chocobozzz/PeerTube/blob/develop/support/docker/production/config/custom-environment-variables.yaml#L14 if the file exists in `NODE_CONFIG_DIR`.

**Comment by decentral1se** [comment]
Thanks for the help! :rocket: 

This was it:

```
export NODE_ENV=production
export NODE_CONFIG_DIR=/config
npm run prune-storage
```

**Comment by lucacalcaterra** [comment]
> export NODE_ENV=production
> export NODE_CONFIG_DIR=/config
> npm run prune-storage

Running the above command get:

> peertube@6.0.3 prune-storage
> node ./dist/scripts/prune-storage.js

/app/node_modules/config/lib/config.js:179
    throw new Error('Configuration property "' + property + '" is not defined');
          ^

Error: Configuration property "database.pool.max" is not defined
    at Config.get (/app/node_modules/config/lib/config.js:179:11)
    at file:///app/dist/core/initializers/config.js:27:25
    at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:336:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:106:12)

Node.js v18.19.1
npm notice 
npm notice New minor version of npm available! 10.2.4 -> 10.5.0
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.5.0
npm notice Run npm install -g npm@10.5.0 to update!
npm notice
